### PR TITLE
Feature/decode search url param

### PR DESCRIPTION
**Pull request by ChrysmOre**
### What

Added URL decoding to search parameter for GET /areas

Slightly refactored QueryParams and request objects to be in correct packages

### How to review

Check changes make sense

### Who can review

Anyone

### CODE REVIEW

File: contract/areas.go [hunk 1]
```diff
@@ -9,11 +15,23 @@ type Area struct {
 
 // GetAreasRequest defines the schema for the GET /areas query parameter
 type GetAreasRequest struct {
-	Dataset  string `schema:"dataset"`
-	AreaType string `schema:"area_type"`
+	QueryParams
 	Category string `schema:"q"`
```
Comment (saminahbab) [inline contract/areas.go:19]: is this tag for the struct supposed to be "schema:q"? if so what does this represent? 
I would have assumed some json tag instead

File: handler/request_test.go [hunk 2]
```diff
@@ -4,43 +4,23 @@ import (
...
-
 	Convey("Given a query param that is not in data structure", t, func() {
 		req, err := http.NewRequest("GET", "http://localhost/areas?offset=4&limit=10&X=test", nil)
 		So(err, ShouldBeNil)
-		var ar areaRequest
-		parseRequest(req, &ar)
-		So(ar, ShouldResemble, areaRequest{
-			Limit:    10,
-			Offset:   4,
+		var ar contract.GetAreasRequest
```
Comment (saminahbab) [inline handler/request_test.go:16]: for consistency I would rename to req

Comment (ChrysmOre) [reply handler/request_test.go:16]: The variable was already called `ar`, it's just the struct has moved package. There's already another variable called `req`.

File: contract/areas.go (hunk 1 above)
Comment (ChrysmOre) [reply contract/areas.go:19]: This isn't something I've changed, but the `schema` tag is for automatically decoding query params using the `gorilla/schema` package.

**Review by saminahbab — APPROVED** [review]
LGTM

**Pull request merged**